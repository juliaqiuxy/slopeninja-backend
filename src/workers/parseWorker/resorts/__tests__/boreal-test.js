import fs from 'fs';
import {
  parseBorealSnow,
  parseBorealLiftCounts,
  parseBorealTrailCounts,
  parseBorealLifts,
  parseBorealTrails,
} from '../boreal';
import { createJSONParser, decodeEntities } from '../../parserFactory';

test('fetches Boreal snow data correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/boreal-weather.json.input`);

  const resortData = await createJSONParser('snow', parseBorealSnow, decodeEntities)(jsonText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: 8,
      snowDepthBase: 130,
      snowDepthSummit: 135,
    },
  });
});

test('fetches all null for nonexisting values', async () => {
  const resortData = await createJSONParser('snow', parseBorealSnow, decodeEntities)('{}');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches all null for no data', async () => {
  const resortData = await createJSONParser('snow', parseBorealSnow, decodeEntities)();
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Boreal lifts data correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/boreal-weather.json.input`);
  const resortData = await createJSONParser('liftCounts', parseBorealLiftCounts, decodeEntities)(jsonText);
  expect(resortData).toEqual({
    liftCounts: {
      total: 8,
      open: 7,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createJSONParser('liftCounts', parseBorealLiftCounts, decodeEntities)('{}');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Boreal trails data correctly', async () => {
  const jsonText = fs.readFileSync(`${__dirname}/__fixtures__/boreal-weather.json.input`);
  const resortData = await createJSONParser('trailCounts', parseBorealTrailCounts, decodeEntities)(jsonText);
  expect(resortData).toEqual({
    trailCounts: {
      total: 34,
      open: 28,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createJSONParser('trailCounts', parseBorealTrailCounts, decodeEntities)('{}');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Boreal lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/boreal-lifts.json.input`);
  const resortData = await createJSONParser('lifts', parseBorealLifts, decodeEntities)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createJSONParser('lifts', parseBorealLifts, decodeEntities)('{}');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Boreal trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/boreal-trails.json.input`);
  const resortData = await createJSONParser('trails', parseBorealTrails, decodeEntities)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createJSONParser('trails', parseBorealTrails, decodeEntities)('{}');
  expect(resortData).toMatchObject({ trails: [] });
});
