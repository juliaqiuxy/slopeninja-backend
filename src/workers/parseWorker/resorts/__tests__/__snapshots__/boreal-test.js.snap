// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Boreal lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": "Accelorator Zone",
      "name": "Accelerator Express Quad Charilift",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "name": "California Cruiser Quad Chairlift",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "name": "Castle Peak Quad Chairlift",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "name": "Lil Carpet",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "name": "Mountain Explorer",
      "status": "open",
    },
    Object {
      "category": "49er Zone",
      "name": "49er Triple Chairlift",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "name": "Cedar Ridge Triple Chairlift",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "name": "Dutchman",
      "status": "open",
    },
  ],
}
`;

exports[`fetches Boreal trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Bonanza",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "His",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Kangaroo Cornice",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Klondike",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "beginner",
      "name": "Lower Cut",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Prospector",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Race Course",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Road Island",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "advanced",
      "name": "Sierra",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "beginner",
      "name": "Sunrise Blvd",
      "status": "open",
    },
    Object {
      "category": "Accelorator Zone",
      "level": "beginner",
      "name": "Sunset Blvd",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "level": "beginner",
      "name": "Adventure Trail",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "level": "beginner",
      "name": "Kiss A Bear",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "level": "beginner",
      "name": "Murphy's Way",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "level": "beginner",
      "name": "Rick's Return",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "level": "beginner",
      "name": "South 40",
      "status": "open",
    },
    Object {
      "category": "Beginner Zone",
      "level": "beginner",
      "name": "Walt's Run",
      "status": "open",
    },
    Object {
      "category": "49er Zone",
      "level": "advanced",
      "name": "Chukker",
      "status": "open",
    },
    Object {
      "category": "49er Zone",
      "level": "advanced",
      "name": "Mogul Rock",
      "status": "open",
    },
    Object {
      "category": "49er Zone",
      "level": "advanced",
      "name": "Ptarmigam",
      "status": "open",
    },
    Object {
      "category": "49er Zone",
      "level": "advanced",
      "name": "Waterfall",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Cedar Return",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Hers",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Juniper",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Mariposa",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Ponderosa",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Tamarack",
      "status": "open",
    },
    Object {
      "category": "Cedar Ridge Zone",
      "level": "advanced",
      "name": "Yours",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "level": "advanced",
      "name": "Blizzard Ridge",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "level": "advanced",
      "name": "Central Pacific",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "level": "advanced",
      "name": "D. & R. G.",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "level": "advanced",
      "name": "Dutchman Gully",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "level": "advanced",
      "name": "Undertow",
      "status": "open",
    },
    Object {
      "category": "Flying Dutchman Zone",
      "level": "advanced",
      "name": "Westward Ho",
      "status": "open",
    },
  ],
}
`;
